File: app/src/main/java/io/jeffchang/buttondemo/network/remote/FakeButtonService.java
package io.jeffchang.buttondemo.network.remote;


import java.util.List;

import io.jeffchang.buttondemo.models.User;
import io.reactivex.Single;
import retrofit2.http.Field;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

/**
 * Service that defines the RESTful interactions with the API.
 */

public interface FakeButtonService {

    @POST("user")
    Single<User> postUsers(@Field("name") String name,
                           @Field("email") String user,
                           @Field("candidate") String candidateId);

    @GET("user")
    Single<List<User>> getUsers(@Query("candidate") String candidate);

}

File: app/src/main/java/io/jeffchang/buttondemo/network/remote/FakeButtonService.kt
package io.jeffchang.buttondemo.network.remote


import io.jeffchang.buttondemo.models.User
import io.reactivex.Single
import retrofit2.http.*

/**
 * Service that defines the RESTful interactions with the API.
 */

interface FakeButtonService {

    @POST("user")
    fun postUser(@Body user: User): Single<User>

    @GET("user")
    fun getUsers(@Query("candidate") candidate: String): Single<List<User>>

    companion object {
        const val CANDIDATE_ID = "vbnguirofbm34!"
    }
}
